google/cloud/sql/connector/utils.py

```diff
@@ -14,13 +14,25 @@
 limitations under the License.
 """
 
+import asyncio
+from functools import partial, wraps
 
 import pymysql.cursors
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
 from cryptography.hazmat.primitives.asymmetric import rsa
 
 
+def run_function_as_async(func):
+    @wraps(func)
+    async def wrapped_sync_function(*args, **kwargs):
+        partial_func = partial(func, *args, **kwargs)
+        loop = asyncio.get_event_loop()
```

Comment:
one thing we need to be careful here is that I think each ICM has a different asyncio loop being used. Are we sure this is going to be getting the right one?

Reply:
I think because the function was called with loop.create_task it would get the right loop, but just to be sure, I updated this in the last commit to explicitly pass in a loop in kwargs